widget/class-widget.php

```diff
@@ -0,0 +1,229 @@
+         <label for="<?php echo esc_attr( $this->get_field_id( 'list_type_radio_btn_1' ) ); ?>"><?php _e( 'Flexible', 'wired-impact-volunteer-management' ); ?></label>
+         <input id="<?php echo esc_attr( $this->get_field_id( 'list_type_radio_btn_2' ) ); ?>" name="<?php echo esc_attr( $this->get_field_name( 'list_type_radio_btn' ) ); ?>" type="radio" value="one-time" <?php if( $list_type === 'one-time' ) { echo 'checked'; } ; ?>/>
+         <label for="<?php echo esc_attr( $this->get_field_id( 'list_type_radio_btn_2' ) ); ?>"><?php _e( 'One-Time', 'wired-impact-volunteer-management' ); ?></label>
+      </p>
+
+      <p>
+         <label class="wi-widget-block-label"><?php _e( 'Number of opportunites to show:', 'wired-impact-volunteer-management'); ?></label>
+         <input id="<?php echo esc_attr( $this->get_field_id( 'number_of_opps_input_1' ) ); ?>" name="<?php echo esc_attr( $this->get_field_name( 'number_of_opps_input' ) ); ?>" type="number" value="<?php echo $num_of_opps; ?>"/>
+      </p>
+
+      <p>
+         <label class="wi-widget-block-label"><?php _e( 'Information to show for each opportunity (in addition to title):', 'wired-impact-volunteer-management'); ?></label>
+         <input id="<?php echo esc_attr( $this->get_field_id( 'opp_info_cbx_when' ) ); ?>" name="<?php echo esc_attr( $this->get_field_name( 'opp_info_when' ) ); ?>" type="checkbox" value="when" <?php if( isset( $instance['opp_info_when'] ) ) { echo 'checked'; }; ?>/>
+         <label for="<?php echo esc_attr( $this->get_field_id( 'opp_info_cbx_when' ) ); ?>" class="wi-widget-block-label"><?php _e( 'When', 'wired-impact-volunteer-management' ); ?></label>
+         <input id="<?php echo esc_attr( $this->get_field_id( 'opp_info_cbx_where' ) ); ?>" name="<?php echo esc_attr( $this->get_field_name( 'opp_info_where' ) ); ?>" type="checkbox" value="where" <?php if( isset( $instance['opp_info_where'] ) ) { echo 'checked'; }; ?>/>
+         <label for="<?php echo esc_attr( $this->get_field_id( 'opp_info_cbx_where' ) ); ?>" class="wi-widget-block-label"><?php _e( 'Where', 'wired-impact-volunteer-management' ); ?></label>
+         <input id="<?php echo esc_attr( $this->get_field_id( 'opp_info_cbx_spots' ) ); ?>" name="<?php echo esc_attr( $this->get_field_name( 'opp_info_spots' ) ); ?>" type="checkbox" value="spots" <?php if( isset( $instance['opp_info_spots'] ) ) { echo 'checked'; }; ?>/>
+         <label for="<?php echo esc_attr( $this->get_field_id( 'opp_info_cbx_spots' ) ); ?>" class="wi-widget-block-label"><?php _e( 'Open Volunteer Spots', 'wired-impact-volunteer-management' ); ?></label>
+      </p>
+
+      <p>
+         <label class="wi-widget-block-label"><?php _e( 'Provide page slug to link to all opportunities of chosen type:', 'wired-impact-volunteer-management'); ?></label>
```

Comment:
This is the part I struggle with the most from a user experience perspective. I see a lot of people not knowing what a slug is, and even if they do, I see them not knowing what to copy and paste here. Is it the full URL, or is it what's after the domain?

I think we can take one of two approaches:

1. We can provide an example URL using their domain so they can see what we're looking, as well as change the word slug to be more clear.
2. We can remove this field entirely and based on whether they select one-time or flexible, we find the page with that shortcode ourselves and link it automatically. That said, this requires a database query, and I would put this out of scope for this project, so no need to move forward with this one.

Feel free to write back here if you want to discuss this a little more.